### DISCUSSION TITLE
Can I use Authenticator for my github account?

### DISCUSSION THREAD

**Discussion by rolfTurner** [Authenticator-Extension/Authenticator#739] [Q&A]
The instructions for configuring two factor authentication for github accounts seem to require the use of a cell phone.  I don't have a cell phone; I *refuse* to have a cell phone!

I use/have to use  two factor authentication with a couple of other accounts that I
have.  Both of these allow me to effect two factor authentication via the "Authenticator" extension on my web browser (I use Firefox).  I simply have to click on the "Authenticator" button on a bar near the top of my browser.  No cell phone needed.

Can I configure two factor authentication for my github account so as to make use of "Authenticator"?  If so, how do I do this?  Please keep your instructions simple,  if you would be so kind!   Assume little to no background knowledge.  I am a Bear of Very Little Brain! :-)

**Comment by mymindstorm** [comment]
You should be able to use this guide: https://authenticator.cc/docs/en/quickstart

Just proceed through the prompts on GitHub as if the extension were the app on your phone.

**Comment by rolfTurner** [comment]
On Sun, 12 Sep 2021 17:33:25 -0700
mymindstorm ***@***.***> wrote:

> You should be able to use this guide:
> https://authenticator.cc/docs/en/quickstart

<SNIP>

Thanks very much for this.  Looks good, except that I came to a
shuddering halt at the instruction to input "the account secret given
to you in the Secret field".  By whom/what was "the account secret"
given to me?  When?  Where?  How?  What *is* "the account secret"?
The example shows it to be a string of capital letters and digits.
(Some sort of encryption key?)  Where do I actually get it from?

Sorry for being a thicko.

cheers,

Rolf Turner

-- 
Honorary Research Fellow
Department of Statistics
University of Auckland
Phone: +64-9-373-7599 ext. 88276

**Reply by mymindstorm** [reply]
The 'secret' value corresponds to the 'text code' referred to by this GitHub support page: https://docs.github.com/en/github/authenticating-to-github/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication

'TOTP app' would be the extension in this case.

It should be easier to just scan the QR code

**Reply by mymindstorm** [reply]
To clarify, you can add an account by scanning a QR code OR manually inputting the 'secret' text.

**Reply by FitaBiskitZzz** [reply]
Hi Sorry if this is a stupid question but  I do not know how to set it this manually, where do I grab the secret account or secret option in the 2FA app? and how to send donations too to your team?

**Comment by rolfTurner** [comment]
On Sun, 12 Sep 2021 21:11:31 -0700
mymindstorm ***@***.***> wrote:

> The 'secret' value corresponds to the 'text code' referred to by this
> GitHub support page:
> https://docs.github.com/en/github/authenticating-to-github/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication

> 'TOTP app' would be the extension in this case.

I couldn't see anywhere to specify this "extension".

> It should be easier to just scan the QR code

Actually I found that I *couldn't* scan the QR code.  One flashed up
briefly, but disappeared before I could take a screen shot of it,
and then never came back again.

After some struggle I managed to get a "text code", and put that into
the "Secret" slot.  But it didn't work.  My account still says "Two
factor authentication is not enabled yet."

Thanks for your efforts, but they seem to have been in vain.

cheers,

Rolf Turner

-- 
Honorary Research Fellow
Department of Statistics
University of Auckland
Phone: +64-9-373-7599 ext. 88276

**Reply by Sneezry** [reply]
I made a recording here https://youtu.be/S_hKidSflas

**Reply by Sneezry** [reply]
I enabled auto fill feature. If you don't, remember to paste the code into the box yourself

![image](https://user-images.githubusercontent.com/1621293/133042628-dc2f185b-28c9-4830-ba5f-b8b03cceb97c.png)

**Comment by rolfTurner** [comment]
On Mon, 13 Sep 2021 00:33:09 -0700
Zhe Li ***@***.***> wrote:

> I made a recording here
> https://youtu.be/S_hKidSflas
> 

Got it.  Managed to follow your steps and now my github account says
"Two-factor authentication" is "Enabled".  So I guess it worked.
The steps that I was missing were clicking on the github rectangle in
the Authenticator menus, pasting the resulting number into the
appropriate rectangle on the configuration screen, and then clicking on
"Download".  This was not at all intuitive to me, and I would never
have got there without your guidance.

Thank you hugely!!!

However I still cannot clone a repository.  When I issue a clone
command in a terminal window, I (still) get prompted for a password,
and then get told "Support for password authentication was removed on
August 13, 2021. Please use a personal access token instead."

So how do I "use a personal access token"? I've tried copying and
pasting in the 6 digit number that I get from Authenticator, but that
instigates the same response/error message.

Any thoughts?

Thanks very much.

cheers,

Rolf Turner

-- 
Honorary Research Fellow
Department of Statistics
University of Auckland
Phone: +64-9-373-7599 ext. 88276

**Comment by Sneezry** [comment]
See
***@***.***/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token

Use personal access token as password for git clone.

rolfTurner ***@***.***>于2021年9月14日 周二上午5:23写道：

> On Mon, 13 Sep 2021 00:33:09 -0700

**Comment by rolfTurner** [comment]
On Mon, 13 Sep 2021 15:29:30 -0700
Zhe Li ***@***.***> wrote:

> See
> ***@***.***/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token
> 
> Use personal access token as password for git clone.

Most of the foregoing is completely incomprehensible.  Gold medal for
cryptic terseness.

However I managed through some web searching to findC instructions for "creating-a-personal-access-token".

Then I managed to create one.  Using it was another matter.

Since the "token" is so incredibly long and complex, I thought I'd try
to follow the instructions for "Caching your GitHub credentials in Git".
This required installing "gh" which I managed to do, although the
instructions in the github docs did not work.  Some web searching led
me to some instructions that I did manage to get to work.  I then
did "gh auth login" as instructed, followed the prompts, and after
quite a bit of thrashing around got to an announcement that I had
succeeded.

But I hadn't.  Trying to clone a repository still resulted in my being
asked for a password, and when I entered my password I was told that
this was no longer acceptable.

Finally I used my mouse to copy the unbelieveably complex "token" and
paste it in (*instead* of my password) when promted for a password.
Mirabile dictu, that worked.  And I succeeded in cloning the repository
that I wanted.

But for God's sake!  Why does this have to be so ridiculously
complicated and why do the instructions have to be so totally
incomprehensible and cryptic?

I've spent about two days fighting with this issue, and seem to have
achieved a successful result more by luck and stubborn persistance than
anything else.

Can't you guys do any better?  Have a little consideration for ordinary
users who are not gurus and who do not have 100 million acronyms and
bits of technical jargon memorised!!!

cheers,

Rolf Turner

-- 
Honorary Research Fellow
Department of Statistics
University of Auckland
Phone: +64-9-373-7599 ext. 88276

**Comment by Sneezry** [comment]
Hi Rolf,

I'm sorry to know the bad experience you had suffered.

But this project is simply an open source 2FA client. It implements totp
and hotp algorithm.

We have no relationship with GitHub or git. We do not provide support for
GitHub or git issues. You may ask GitHub to improve their docs or their
service user experience.

Regards,
Zhe

rolfTurner ***@***.***>于2021年9月14日 周二上午9:30写道：

> On Mon, 13 Sep 2021 15:29:30 -0700

**Comment by rolfTurner** [comment]
On Mon, 13 Sep 2021 18:42:00 -0700
Zhe Li ***@***.***> wrote:

> Hi Rolf,
> 
> I'm sorry to know the bad experience you had suffered.
> 
> But this project is simply an open source 2FA client. It implements
> totp and hotp algorithm.
> 
> We have no relationship with GitHub or git. We do not provide support
> for GitHub or git issues. You may ask GitHub to improve their docs or
> their service user experience.
> 
> Regards,
> Zhe

Sorry for "venting" at you.  It's not your fault/responsibility. I was
feeling extremely peeved and frustrated, and just *had* to vent at
somebody.

However the situation you describe:

"But this project is simply an open source 2FA client. It implements
totp and hotp algorithm. We have no relationship with GitHub or git."

is a large part of why I am so confused and frustrated.  There is no
obvious structure, no clear indication of who to contact for help,
no clear connection from anything to anything else, no clear
delineation of who has responsibility for what.  There is no way to
be clear who I am actually communicating with! I guess I'm venting
again, and I repeat, it's not your fault or responsibility.

Moreover you were a great help to me, and guided me through to a
working solution to my problem, even though there were aspects of
the with which I am still unhappy.

Thanks again.

cheers,

Rolf Turner

-- 
Honorary Research Fellow
Department of Statistics
University of Auckland
Phone: +64-9-373-7599 ext. 88276
